github/repos.go

```diff
@@ -1037,6 +1037,33 @@ func (s *RepositoriesService) getBranchFromURL(ctx context.Context, u string, fo
 	return resp, err
 }
 
+// RenameBranchRequest represents a request to rename a branch.
+type RenameBranchRequest struct {
+	NewName string `json:"new_name"`
+}
+
+// RenameBranch renames a branch in a repository.
+//
+// To rename a non-default branch: Users must have push access. GitHub Apps must have the `contents:write` repository permission.
+// To rename the default branch: Users must have admin or owner permissions. GitHub Apps must have the `administration:write` repository permission.
+//
+// GitHub API docs: https://docs.github.com/en/rest/reference/repos#rename-a-branch
+func (s *RepositoriesService) RenameBranch(ctx context.Context, owner, repo, branch string, renameBranchReq *RenameBranchRequest) (*Branch, *Response, error) {
```

Comment:
I'm thinking of simplifying the API a bit.

How about this:

```suggestion
func (s *RepositoriesService) RenameBranch(ctx context.Context, owner, repo, branch, newName string) (*Branch, *Response, error) {
```

and you can keep your new struct, but you no longer would export it, like this:

```
type renameBranchRequest struct {
	NewName string `json:"new_name"`
}
```

and then just use it within the method by creating a new one on demand:

```
r := &renameBranchRequest{NewName: newName}
req, err := s.client.NewRequest("POST", u, r)
```

Reply:
@gmlewis, thank you.
Please take a look at https://github.com/google/go-github/pull/2119/commits/c9e1ab682b7241b975aa7f3a0361592ab57e7660